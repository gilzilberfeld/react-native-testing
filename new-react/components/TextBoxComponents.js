import { Platform } from "react-native";

import { Text, View } from "./Themed";

export function StaticMonoText(props) {
  return (
    <Text 
      {...props}
      style={[
        props.style,
        {
          // The "code" font is different on each platform.
          fontFamily: Platform.select({
            default: "Courier",
            ios: "Courier New",
            android: "monospace",
          }),
          fontWeight: "500",
        },
      ]}
    >  
      StaticText
    </Text>
  )
};

export function ParametericMonoText(props) {
  return (
    <Text 
      {...props}
      style={[
        props.style,
        {
          // The "code" font is different on each platform.
          fontFamily: Platform.select({
            default: "Courier",
            ios: "Courier New",
            android: "monospace",
          }),
          fontWeight: "500",
        },
      ]}
    >  
      {props}
    </Text>
  )
};
