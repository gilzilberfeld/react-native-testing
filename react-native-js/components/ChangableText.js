import { Platform } from "react-native";
import { Text, View } from "./Themed";

async function setText(){
 res =  await fetch('https://api.amazon.com/')
            .then ( result => result.json());
  textToShow = res.amazonData['product'];
};


export var textToShow = "Before";
export function ChangableText(props) {
  return (
    <Text 
      {...props}
      style={[
        props.style,
        {
          // The "code" font is different on each platform.
          fontFamily: Platform.select({
            default: "Courier",
            ios: "Courier New",
            android: "monospace",
          }),
          fontWeight: "500"
        },
        
      ]} onPress={setText}
    >  
    {textToShow}
    </Text>
  )
};
