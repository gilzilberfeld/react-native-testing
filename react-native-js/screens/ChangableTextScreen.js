import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { ChangableText } from "../components/ChangableText";

export default function ChangableTextScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <ChangableText testID="changableText" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
    padding: 20,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  link: {
    marginTop: 15,
    paddingVertical: 15,
  },
  linkText: {
    fontSize: 14,
    color: "#2e78b7",
  },
});
