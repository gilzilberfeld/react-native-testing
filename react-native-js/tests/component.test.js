import {fireEvent, render,  } from '@testing-library/react-native'
import {StaticMonoText, ChangableText, textToShow} from '../components/TextBoxComponents'
import EditScreenInfo from '../components/EditScreenInfo'

import '@testing-library/jest-dom'
import ChangableTextScreen from '../screens/ChangableTextScreen';

describe ('component tests example suite', ()=>{
    test('render to JSON', ()=>{
        const { toJSON } = render(<StaticMonoText/>); 
        expect(toJSON()).toMatchSnapshot();
    })

    test('getbyText example', ()=> {
        const { getAllByRole, getByRole , findByRole, getByTestId, getByText} = render(<EditScreenInfo/>);
        const elementText = getByText("InnerText");
        expect(elementText.props.children).toEqual('InnerText');
    })


    // note that there's only one Text element with AccessibilityRole
    test('getAllbyRole example', ()=> {
        const { getAllByRole, getByRole , findByRole, getByTestId, getByText} = render(<EditScreenInfo/>);
        const elementText = getAllByRole("text")[0];
        expect(elementText.props.children).toEqual('InnerText');
    })

    test('getAllbyRole example', ()=> {
        const { getAllByRole, getByRole , findByRole, getByTestId, getByText} = render(<EditScreenInfo/>);
        const elementText = getAllByRole("text")[0];
        expect(elementText.props.children).toEqual('InnerText');
    })

    test('getByTestID example', ()=> {
        const { getAllByRole, getByRole , findByRole, getByTestId, getByText} = render(<EditScreenInfo/>);
        const elementText = getByTestId("theTestID");
        expect(elementText.props.children).toEqual('UpperText');
    })

    it('checking internal component', ()=>{
        const { getAllByRole, getByRole , findByRole, getByTestId, getByText} = render(<EditScreenInfo/>);
        const theStaticMonoTextElement = getByTestId("theMonoTestID");
        expect(theStaticMonoTextElement.props.children).toEqual('StaticText');
    })

    beforeEach(()=>{
        fetch.resetMocks();
    });

    it ('mock fetch and wait for promise to end', async()=>{
        const {getByTestId, update} = render(<ChangableTextScreen/>); 
        theText = getByTestId("changableText");
        expect(theText.props.children).toEqual("Before");
        fetch.mockResponseOnce(JSON.stringify(
            {
                "amazonData":   {
                    "product": "theProduct"
                }
            }));
        
        await fireEvent.press(theText);  
        expect(textToShow).toEqual("theProduct");
    })    
});
