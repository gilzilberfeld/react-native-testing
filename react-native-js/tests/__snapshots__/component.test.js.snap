// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component tests example suite render to JSON 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "Courier New",
          "fontWeight": "500",
        },
      ],
    ]
  }
>
  StaticText
</Text>
`;
